test/receiverSpec.js

```diff
@@ -529,18 +529,43 @@ describe('Receiver Module', function () {
           const kep = this.receiver.generateSharedSecret()
           const request = sender.createRequest(Object.assign({}, kep, {
             destination_amount: "1",
+            data: { for: 'that thing' }
           }))
 
           this.transfer.account = request.address
           this.transfer.executionCondition = request.condition
           this.transfer.data.ilp_header.data.expires_at = request.expires_at
           this.transfer.data.ilp_header.account = request.address
-          delete this.transfer.data.ilp_header.data.data
+          this.transfer.data.ilp_header.data.data = request.data
 
           yield expect(this.client.emitAsync('incoming_prepare', this.transfer))
             .to.be.rejectedWith('opts.reviewPayment was not specified, so KEP is not supported on this receiver')
         })
 
+        it('should not fulfill KEP payment with invalid ciphertext blob', function * () {
+          const sender = createSender({
+            client: new MockClient({}),
+            uuidSeed: Buffer.from('f73e2739c0f0ff4c9b7cac6678c89a59ee6cb8911b39d39afbf2fef9e77bc9c3', 'hex')
+          })
+
+          const kep = this.receiver.generateSharedSecret()
+          const request = sender.createRequest(Object.assign({}, kep, {
+            destination_amount: "1",
+            data: { for: 'that thing' }
+          }))
+
+          this.transfer.account = request.address
+          this.transfer.executionCondition = request.condition
+          this.transfer.data.ilp_header.data.expires_at = request.expires_at
+          this.transfer.data.ilp_header.account = request.address
+          this.transfer.data.ilp_header.data.data = {
```

Comment:
`transfer.data.ilp_header.data.data.blob` is absurd. Can we please switch to the new packet format?? @justmoon